ServerCore/Pages/Puzzles/Checklist.cshtml.cs

```diff
@@ -73,7 +79,28 @@ public async Task<IActionResult> OnGetAsync()
                 puzzleAnswers.TryGetValue(puzzle.ID, out ContentFile puzzleAnswer);
                 puzzlePrereqs.TryGetValue(puzzle.ID, out List<string> prereqs);
                 puzzleResponses.TryGetValue(puzzle.ID, out ResponseData responses);
-                puzzleHints.TryGetValue(puzzle.ID, out int hints);
+                puzzleHints.TryGetValue(puzzle.ID, out IEnumerable<int> hints);
+
+                int totalHintCostThisPuzzle = 0;
+                int hintsCountThisPuzzle = 0;
+                if (hints != null)
+                {
+                    int minHintCostThisPuzzle = 0;
+
+                    hintsCountThisPuzzle = hints.Count();
+                    foreach (int cost in hints)
+                    {
+                        minHintCostThisPuzzle = Math.Min(minHintCostThisPuzzle, cost);
+                    }
+
+                    totalHintCostThisPuzzle = Math.Abs(minHintCostThisPuzzle);
+
+                    foreach (int cost in hints)
+                    {
+                        totalHintCostThisPuzzle += Math.Max(0, cost + minHintCostThisPuzzle);
```

Comment:
This code confused me at first but I think will compute the correct value. I think I would include this comment:
```
// Negative cost hints will be ignored because Max(0,negative) is 0.
// Positive cost hints will only be counted for the cost above the discount.
// (The discount is counted against each other hint.)
```